# Women Shoes Data Analysis

In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('7210_1.csv')
df.head()

C:\Users\Rocky\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (13,16,18,22,24,32,33,40,41,47,51) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


id asins   brand  \
0  AVpe__eOilAPnD_xSt-H   NaN  Novica   
1  AVpe__eOilAPnD_xSt-H   NaN  Novica   
2  AVpe__eOilAPnD_xSt-H   NaN  Novica   
3  AVpe__eOilAPnD_xSt-H   NaN  Novica   
4  AVpe__eOilAPnD_xSt-H   NaN  Novica   

                                categories  colors  count  \
0  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
1  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
2  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
3  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   
4  Access.,Clothing,Shoes,Women's Clothing  Purple    NaN   

              dateAdded           dateUpdated  \
0  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
1  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
2  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
3  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   
4  2017-02-03T22:06:24Z  2017-03-28T11:53:51Z   

                                        descriptions dimension  ...  \
0  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
1  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
2  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
3  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   
4  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...       NaN  ...   

                                                skus  \
0  [{"sourceURLs":["https://www.overstock.com/Wor...   
1  [{"sourceURLs":["https://www.overstock.com/Wor...   
2  [{"sourceURLs":["https://www.overstock.com/Wor...   
3  [{"sourceURLs":["https://www.overstock.com/Wor...   
4  [{"sourceURLs":["https://www.overstock.com/Wor...   

                                          sourceURLs  upc vin  websiteIDs  \
0  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
1  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
2  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
3  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   
4  https://www.overstock.com/Worldstock-Fair-Trad...  NaN NaN         NaN   

  weight Unnamed: 48 Unnamed: 49 Unnamed: 50 Unnamed: 51  
0    NaN         NaN         NaN         NaN         NaN  
1    NaN         NaN         NaN         NaN         NaN  
2    NaN         NaN         NaN         NaN         NaN  
3    NaN         NaN         NaN         NaN         NaN  
4    NaN         NaN         NaN         NaN         NaN  

[5 rows x 52 columns]

In [3]:
df.shape

(33801, 52)

In [4]:
df.isnull().sum()

id                         0
asins                  31379
brand                    984
categories                 0
colors                 13957
count                  33801
dateAdded                  0
dateUpdated                0
descriptions            5376
dimension              32624
ean                    24023
features                6620
flavors                33801
imageURLs              11492
isbn                   33801
keys                       0
manufacturer           24522
manufacturerNumber      2367
merchants              18419
name                       0
prices.amountMin           0
prices.amountMax           0
prices.availability    33494
prices.color           16380
prices.condition       21786
prices.count           33801
prices.currency            0
prices.dateAdded           0
prices.dateSeen            0
prices.flavor          33801
prices.isSale              0
prices.merchant         3875
prices.offer           27003
prices.returnPolicy    32427
prices.shippin

In [5]:
#Removing all the columns with all null values
df.dropna(how="all", axis=1, inplace=True)
df.head()

id asins   brand  \
0  AVpe__eOilAPnD_xSt-H   NaN  Novica   
1  AVpe__eOilAPnD_xSt-H   NaN  Novica   
2  AVpe__eOilAPnD_xSt-H   NaN  Novica   
3  AVpe__eOilAPnD_xSt-H   NaN  Novica   
4  AVpe__eOilAPnD_xSt-H   NaN  Novica   

                                categories  colors             dateAdded  \
0  Access.,Clothing,Shoes,Women's Clothing  Purple  2017-02-03T22:06:24Z   
1  Access.,Clothing,Shoes,Women's Clothing  Purple  2017-02-03T22:06:24Z   
2  Access.,Clothing,Shoes,Women's Clothing  Purple  2017-02-03T22:06:24Z   
3  Access.,Clothing,Shoes,Women's Clothing  Purple  2017-02-03T22:06:24Z   
4  Access.,Clothing,Shoes,Women's Clothing  Purple  2017-02-03T22:06:24Z   

            dateUpdated                                       descriptions  \
0  2017-03-28T11:53:51Z  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
1  2017-03-28T11:53:51Z  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
2  2017-03-28T11:53:51Z  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
3  2017-03-28T11:53:51Z  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
4  2017-03-28T11:53:51Z  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   

  dimension  ean  ... prices.size  \
0       NaN  NaN  ...           L   
1       NaN  NaN  ...           M   
2       NaN  NaN  ...           S   
3       NaN  NaN  ...          XL   
4       NaN  NaN  ...          XL   

                                   prices.sourceURLs  \
0  https://www.overstock.com/Worldstock-Fair-Trad...   
1  https://www.overstock.com/Worldstock-Fair-Trad...   
2  https://www.overstock.com/Worldstock-Fair-Trad...   
3  https://www.overstock.com/Worldstock-Fair-Trad...   
4  https://www.overstock.com/Worldstock-Fair-Trad...   

                                          quantities  \
0  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
1  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
2  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
3  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   
4  [{"dateSeen":["2017-03-26T05:04:46.539Z","2017...   

                                             reviews sizes  \
0  [{"date":"2017-01-10T00:00:00.000Z","dateAdded...   NaN   
1  [{"date":"2017-01-10T00:00:00.000Z","dateAdded...   NaN   
2  [{"date":"2017-01-10T00:00:00.000Z","dateAdded...   NaN   
3  [{"date":"2017-01-10T00:00:00.000Z","dateAdded...   NaN   
4  [{"date":"2017-01-10T00:00:00.000Z","dateAdded...   NaN   

                                                skus  \
0  [{"sourceURLs":["https://www.overstock.com/Wor...   
1  [{"sourceURLs":["https://www.overstock.com/Wor...   
2  [{"sourceURLs":["https://www.overstock.com/Wor...   
3  [{"sourceURLs":["https://www.overstock.com/Wor...   
4  [{"sourceURLs":["https://www.overstock.com/Wor...   

                                          sourceURLs  upc  weight Unnamed: 51  
0  https://www.overstock.com/Worldstock-Fair-Trad...  NaN     NaN         NaN  
1  https://www.overstock.com/Worldstock-Fair-Trad...  NaN     NaN         NaN  
2  https://www.overstock.com/Worldstock-Fair-Trad...  NaN     NaN         NaN  
3  https://www.overstock.com/Worldstock-Fair-Trad...  NaN     NaN         NaN  
4  https://www.overstock.com/Worldstock-Fair-Trad...  NaN     NaN         NaN  

[5 rows x 40 columns]

In [6]:
df.shape

(33801, 40)

In [7]:
df.dtypes

id                      object
asins                   object
brand                   object
categories              object
colors                  object
dateAdded               object
dateUpdated             object
descriptions            object
dimension               object
ean                    float64
features                object
imageURLs               object
keys                    object
manufacturer            object
manufacturerNumber      object
merchants               object
name                    object
prices.amountMin       float64
prices.amountMax       float64
prices.availability     object
prices.color            object
prices.condition        object
prices.currency         object
prices.dateAdded        object
prices.dateSeen         object
prices.isSale             bool
prices.merchant         object
prices.offer            object
prices.returnPolicy     object
prices.shipping         object
prices.size             object
prices.sourceURLs       object
quantiti

In [8]:
df['dateAdded']=pd.to_datetime(df.dateAdded)
df['dateUpdated']=pd.to_datetime(df.dateUpdated)

In [9]:
df['dateAdded'].dtypes

datetime64[ns, UTC]

In [10]:
#To get the month name from the date time format of pandas

df.dateAdded.dt.month_name()

0        February
1        February
2        February
3        February
4        February
           ...   
33796       March
33797       March
33798       March
33799       April
33800       April
Name: dateAdded, Length: 33801, dtype: object

# 1. What is number of brands of shoes?


In [11]:
df.brand.unique()

array(['Novica', 'Pedi Couture', 'ANNE KLEIN SPORT', ..., 'Prince',
       'PINUP COUTURE', 'Amalfi by Rangoni'], dtype=object)

In [12]:
#number of brands of shoes

len(df.brand.unique())

1361

# 2. What are the number of items in stock?

In [13]:
df['prices.availability'].value_counts()

In Stock        142
TRUE            106
Out Of Stock     47
YES              10
yes               1
FALSE             1
Name: prices.availability, dtype: int64

There are only 142 number of items in stock

# 3. Price of top ten brands of shoes (Sorted volume on the basis of the first acceptance criteria).

In [14]:
df_top10=df.sort_values(by='prices.amountMax', ascending=False).reset_index()
df_top10['prices.amountMax'].head()

0    5200.00
1    4270.00
2    4198.99
3    4198.99
4    3499.00
Name: prices.amountMax, dtype: float64

In [15]:
df_top10.loc[0:9,['brand','prices.amountMax']]

brand  prices.amountMax
0              NaN           5200.00
1              NaN           4270.00
2  JewelsObsession           4198.99
3  JewelsObsession           4198.99
4             Teva           3499.00
5  JewelsObsession           3109.99
6  JewelsObsession           3109.99
7             Teva           2999.00
8              NaN           2694.00
9            Gucci           2300.00

# 4. Number of Black Shoes

In [16]:
df_black=df['prices.color'].str.contains('Black')
df_black.value_counts(dropna=False)

NaN      16380
False    15837
True      1584
Name: prices.color, dtype: int64

There are 1584 back shoes in the database
